### PR TITLE
feat: added skin category

### PR DESCRIPTION
**Pull request by cazala**
Added `skin` category

### CODE REVIEW

File: src/modules/editor/utils.ts [hunk 1]
```diff
@@ -213,6 +214,29 @@ export function mergeWearables(avatar: Wearable[], apply: Wearable[]) {
   return Object.values(wearables)
 }
 
+export function patchWearables(wearables: Wearable[]) {
```
Comment (LautaroPetaccio) [inline src/modules/editor/utils.ts:217]: Would you mind adding a test for this new function?

File: src/components/ItemEditorPage/RightPanel/RightPanel.tsx [hunk 2]
```diff
@@ -109,36 +118,47 @@ export default class RightPanel extends React.PureComponent<Props, State> {
...
   handleChangeCategory = (category: WearableCategory) => {
-    const data = {
+    let data: WearableData = {
       ...this.state.data!,
       category
     }
+    // when changing the category to SKIN we hide everything else
+    if (category === WearableCategory.SKIN) {
+      data = this.setReplaces(data, [])
+      data = this.setHides(data, [])
```
Comment (LautaroPetaccio) [inline src/components/ItemEditorPage/RightPanel/RightPanel.tsx:128]: If the category is sinks, shouldn't we be hiding the categories given by `getSkinHiddenCategories`?

File: src/components/ItemEditorPage/RightPanel/RightPanel.tsx [hunk 3]
```diff
@@ -397,7 +428,7 @@ export default class RightPanel extends React.PureComponent<Props, State> {
                           info={t('item_editor.right_panel.replaces_info')}
                           value={replaces}
                           options={this.asCategorySelect(overrideCategories)}
-                          disabled={!canEditItemMetadata}
+                          disabled={!canEditItemMetadata || this.isSkin()}
```
Comment (LautaroPetaccio) [inline src/components/ItemEditorPage/RightPanel/RightPanel.tsx:431]: Shouldn't the hides selector be disabled as well if it's a skin?

File: src/components/ItemEditorPage/RightPanel/RightPanel.tsx [hunk 4]
```diff
@@ -265,9 +290,15 @@ export default class RightPanel extends React.PureComponent<Props, State> {
...
 
-              const actionableCategories = item ? getOverridesCategories(item.contents) : []
+              let actionableCategories = item ? getOverridesCategories(item.contents) : []
               const wearableCategories = item ? getWearableCategories(item.contents) : []
 
+              if (data?.category === WearableCategory.SKIN) {
+                actionableCategories = actionableCategories.filter(actionableCategory =>
+                  !getSkinHiddenCategories().includes(actionableCategory) && actionableCategory !== WearableCategory.SKIN
+                )
+              }
```
Comment (LautaroPetaccio) [inline src/components/ItemEditorPage/RightPanel/RightPanel.tsx:300]: What if we give the `getOverridesCategories` the item's category as well and put this section of code there?

**Review by LautaroPetaccio — CHANGES_REQUESTED** [review]

File: src/components/ItemEditorPage/RightPanel/RightPanel.tsx (hunk 2 above)
Comment (cazala) [reply src/components/ItemEditorPage/RightPanel/RightPanel.tsx:128]: The categories that a skin hides are not embedded in the wearable, that logic lives in the explorer (all the skins hide the head/upper/lower/feet). What goes in the `hides` list is the extra stuff that this skin hides (ie. the accessories, like `hat`, `eyewear`, etc).

File: src/components/ItemEditorPage/RightPanel/RightPanel.tsx (hunk 3 above)
Comment (cazala) [reply src/components/ItemEditorPage/RightPanel/RightPanel.tsx:431]: No, the creator can choose to hide accessories that are not hidden by default by skins

File: src/components/ItemEditorPage/RightPanel/RightPanel.tsx (hunk 4 above)
Comment (cazala) [reply src/components/ItemEditorPage/RightPanel/RightPanel.tsx]: done ✅

File: src/modules/editor/utils.ts (hunk 1 above)
Comment (cazala) [reply src/modules/editor/utils.ts:222]: done ✅

File: src/modules/editor/utils.ts [hunk 5]
```diff
@@ -213,6 +214,29 @@ export function mergeWearables(avatar: Wearable[], apply: Wearable[]) {
...
+  // makes changes to wearables needed to make them work with the outdated version of client
```
[the remark this answers is not in the document]
Comment (cazala) [reply src/modules/editor/utils.ts]: done ✅

File: src/modules/editor/utils.spec.ts [hunk 6]
```diff
@@ -88,3 +89,66 @@ describe('when converting a catalyst wearable to a wearable', () => {
     expect(fromCatalystWearableToWearable(aCatalystWearable)).toEqual(aWearable)
   })
 })
+
+describe('when patching wearables', () => {
+  describe("and the category of the wearable is 'skin'", () => {
```
Comment (LautaroPetaccio) [inline src/modules/editor/utils.spec.ts:94]: Would you mind adding the other case? The `and the category of the wearable isn't a 'skin'`

**Review by LautaroPetaccio — CHANGES_REQUESTED** [review]

File: src/modules/editor/utils.spec.ts (hunk 6 above)
Comment (cazala) [reply src/modules/editor/utils.spec.ts:121]: Done ✅

**Review by LautaroPetaccio — APPROVED** [review]

**Pull request merged**